### PR TITLE
fix: cdk603 error calculating previousLocalExitRoot

### PR DESCRIPTION
**Pull request by joanestebanr**
## Description

Since `agglayer:0.2.0-rc.15` the responses of ` CertificateHeader` add a new field: `PreviousLocalExitRoot`
- Fixes error calculating  previousLocalExitRoot on new certificate if a certificate is inError

NOTE: **Database incompatibility with previous version!, require to delete it!**

### CODE REVIEW

**Review by ToniRamirezM — APPROVED** [review]

File: agglayer/client_test.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ func TestExploratoryGetEpochConfiguration(t *testing.T) {
 
 func TestExploratoryGetLatestKnownCertificateHeader(t *testing.T) {
 	t.Skip("This test is exploratory and should be skipped")
-	aggLayerClient := NewAggLayerClient("http://localhost:32781")
+	aggLayerClient := NewAggLayerClient("http://localhost:32843")
```
Comment (vcastellm) [inline agglayer/client_test.go:48]: Will this keep changing? should we move this and the previous ones to a constant?

File: agglayer/types.go [hunk 2]
```diff
@@ -574,36 +576,41 @@ func (p *GenericPPError) String() string {
...
+		return nilStr
 	}
 	errors := ""
 	if c.Error != nil {
 		errors = c.Error.String()
 	}
-
-	return fmt.Sprintf("Height: %d, CertificateID: %s, NewLocalExitRoot: %s. Status: %s. Errors: [%s]",
-		c.Height, c.CertificateID.String(), c.NewLocalExitRoot.String(), c.Status.String(), errors)
+	previousLocalExitRoot := "nil"
```
Comment (vcastellm) [inline agglayer/types.go:607]: use the `nilStr` constant?

File: agglayer/types.go [hunk 3]
```diff
@@ -23,6 +23,8 @@ const (
 	Candidate
 	InError
 	Settled
+
+	nilStr = "nil"
```
Comment (vcastellm) [inline agglayer/types.go:27]: Why is this needed, agglayer API?

File: aggsender/aggsender_test.go [hunk 4]
```diff
@@ -283,7 +285,7 @@ func TestAggSenderStart(t *testing.T) {
 		ctx,
 		log.WithFields("test", "unittest"),
 		Config{
-			StoragePath:          "file::memory:?cache=shared",
+			StoragePath:          "file:TestAggSenderStart?mode=memory&cache=shared",
```
Comment (vcastellm) [inline aggsender/aggsender_test.go:288]: Are we 100% sure this is using memory or a file with this name?

File: db/meddler.go [hunk 5]
```diff
@@ -163,6 +163,16 @@ func (b HashMeddler) PostRead(fieldPtr, scanTarget interface{}) error {
 	}
 	field, ok := fieldPtr.(*common.Hash)
 	if !ok {
+		hashPtr, ok := fieldPtr.(**common.Hash)
```
Comment (vcastellm) [inline db/meddler.go:166]: In which case this can be a pointer to the pointer?

Comment (Stefan-Ethernal) [reply db/meddler.go:166]: If I'm not mistaken, this can be the case when the hash value is `NULL` in the DB.

File: agglayer/client_test.go (hunk 1 above)
Comment (Stefan-Ethernal) [reply agglayer/client_test.go:48]: I believe so... Every time Kurtosis is spun up, it assigns different port values. Anyway, this test is skipped, as it is only for some debug/exploratory purposes (and therefore we might want to remove it at some point once the agglayer becomes stable enough).
